Cosmetics.py

```diff
@@ -338,162 +145,387 @@ def patch_navi_colors(rom, settings, log, symbols):
+        colors = {}
+
+        for index, (trail_part, option, rainbow_symbol, color_symbol) in enumerate([
+            ('inner', option_inner, rainbow_inner_symbol, color_inner_symbol),
+            ('outer', option_outer, rainbow_outer_symbol, color_outer_symbol),
+        ]):
+            color = None
+
+            # Plando
+            if len(plando_colors) > 0 and plando_colors[0].get(trail_part, ''):
+                color = hex_to_color(plando_colors[0][trail_part])
 
-        custom_color = False
-        for index, (address, transparency, white_transparency) in enumerate(sword_trail_addresses):
             # set rainbow option
-            if sword_trail_option == 'Rainbow':
-                rom.write_byte(sword_trail_rainbow_symbol, 0x01)
+            if option_inner == 'Rainbow':
+                rom.write_byte(rainbow_symbol, 0x01)
                 color = [0x00, 0x00, 0x00]
-                continue
             else:
-                rom.write_byte(sword_trail_rainbow_symbol, 0x00)
+                rom.write_byte(rainbow_symbol, 0x00)
 
             # handle completely random
-            if sword_trail_option == 'Completely Random':
-                color = [random.getrandbits(8), random.getrandbits(8), random.getrandbits(8)]
-                if sword_trail_name not in log.sword_colors:
-                    log.sword_colors[sword_trail_name] = list()
-                log.sword_colors[sword_trail_name].append(dict(option=sword_trail_option, color=''.join(['{:02X}'.format(c) for c in color[0:3]])))
-
-            elif sword_trail_option in sword_colors:
-                color = list(sword_colors[sword_trail_option][index])
-            # build color from hex code
-            else:
-                color = list(int(sword_trail_option[i:i+2], 16) for i in (0, 2, 4))
-                custom_color = True
+            if color is None and option == 'Completely Random':
+                # Specific handling for inner bombchu trails for contrast purposes.
+                if trail_name == 'Bombchu Trail' and trail_part == 'inner':
+                    fixed_dark_color = [0, 0, 0]
+                    color = [0, 0, 0]
+                    # Avoid colors which have a low contrast so the bombchu ticking is still visible
+                    while contrast_ratio(color, fixed_dark_color) <= 4:
+                        color = generate_random_color()
+                else:
+                    color = generate_random_color()
 
-            if sword_trail_option == 'White':
-                color = color + [white_transparency]
-            else:
-                color = color + [transparency]
+            # grab the color from the list
+            if color is None and option in bombchu_trail_colors:
+                color = list(bombchu_trail_colors[option])
 
-            rom.write_bytes(address, color)
+            # build color from hex code
+            if color is None:
+                color = hex_to_color(option)
```

Comment:
This is throwing an error for Boomerang when option is 'Yellow'. Should that be a bombchu trail option, or should we check boomerang_trail_colors here as well?

Reply:
I pass that in as `trail_color_list`, I guess I just only changed it for random choice and forgot to change it right above those lines to use that instead? Whoops. Fixed, thanks.

Reply:
Disregard, I am bad. Going to fix and force push.

Reply:
Actually fixed now.